collector/lib/CollectorConfig.cpp

```diff
@@ -360,4 +369,60 @@ std::ostream& operator<<(std::ostream& os, const CollectorConfig& c) {
          << ", enable_external_ips:" << c.EnableExternalIPs();
 }
 
+// Returns number of CPUs per buffer for the purpose of ring buffer allocation.
+// The value is adjusted based on the allowed total limit, e.g.:
+//
+// * total limit for ring buffers is 1Gi
+// * one buffer takes 8Mi
+// * maximum allowed number of buffers is 128
+// * total number of CPU cores is 150
+//
+// In this scenario, n_buffers = 150 / 1 > 128, thus the adjusted value will be
+// returned: ceil(150 / 128) = 2. Allocating one buffer per 2 CPU cores will
+// allow us to fit into the total limit: (150 / 2) * 8Mi < 1Gi.
+//
+unsigned int CollectorConfig::GetSinspCpuPerBuffer() const {
+  unsigned int n_buffers, max_n_buffers;
+
+  if (sinsp_buffer_size_ == 0) {
+    CLOG(WARNING) << "Trying to calculate cpu-per-buffer without"
+                     "initialized buffer size. Return raw value.";
```

Comment:
We might want to let users know what that `raw value` is
```suggestion
    CLOG(WARNING) << "Trying to calculate cpu-per-buffer without "
                     "initialized buffer size. Defaulting to " << sinsp_cpu_per_buffer_ << " bytes.";
```